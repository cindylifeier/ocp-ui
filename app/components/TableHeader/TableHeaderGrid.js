import { Grid } from 'styled-css-grid';

const TableHeaderGrid = Grid.extend`
  border-bottom: 1px outset rgb(51, 51, 51);

  &:first-child {
    background-color: #f2f2f2;
    font-size: 20px;
  }

  &:nth-child(odd) {
    background-color: #f2f2f2;
  }
`;

TableHeaderGrid.propTypes = {};

export default TableHeaderGrid;
