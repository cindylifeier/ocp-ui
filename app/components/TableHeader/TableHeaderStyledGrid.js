import { Grid } from 'styled-css-grid';

const TableHeaderStyledGrid = Grid.extend`
  border-bottom: 1px outset rgb(51, 51, 51);

  &:first-child {
    background-color: #f2f2f2;
    font-size: 20px;
  }

  &:nth-child(odd) {
    background-color: #f2f2f2;
  }
`;

TableHeaderStyledGrid.propTypes = {};

export default TableHeaderStyledGrid;
