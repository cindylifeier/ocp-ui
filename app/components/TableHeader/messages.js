/*
 * TableHeader Messages
 *
 * This contains all the text for the TableHeader component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'ocpui.components.TableHeader.header',
    defaultMessage: 'This is the TableHeader component !',
  },
});
