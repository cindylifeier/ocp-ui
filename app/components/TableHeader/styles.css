.rowGridContainer:nth-child(odd) {
  background-color: #f2f2f2;
}

.rowGridContainer:first-child {
  background-color: #f2f2f2;
  font-size: 20px;
}

.rowGridContainer {
  display: grid;
  grid-gap: 5px;
  border-bottom: 1px outset rgb(51, 51, 51);
}
