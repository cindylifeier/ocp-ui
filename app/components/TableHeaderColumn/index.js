/**
 *
 * TableHeaderColumn
 *
 */


import PropTypes from 'prop-types';
import styled from 'styled-components';

const TableHeaderColumn = styled.div`
  color: rgb(51, 51, 51);
  font-family: "Arial Bold", "Arial", sans-serif;
  font-style: normal;
  font-size: 15px;
  font-weight: 700;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-word;
  padding: 2px;
`;

TableHeaderColumn.propTypes = {
  children: PropTypes.node,
};

export default TableHeaderColumn;
