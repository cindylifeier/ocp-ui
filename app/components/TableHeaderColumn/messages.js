/*
 * TableHeaderColumn Messages
 *
 * This contains all the text for the TableHeaderColumn component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'ocpui.components.TableHeaderColumn.header',
    defaultMessage: 'This is the TableHeaderColumn component !',
  },
});
