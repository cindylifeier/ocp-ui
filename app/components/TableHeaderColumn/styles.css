.cellGridHeaderItem {
  color: rgb(51, 51, 51);
  font-family: "Arial Bold", "Arial";
  font-style: normal;
  font-size: 15px;
  font-weight: 700;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-all;
  padding: 2px;
}
