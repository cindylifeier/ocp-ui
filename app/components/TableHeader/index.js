/**
 *
 * TableHeader
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.css';

// import styled from 'styled-components';

function TableHeader({ children }) {
  return (
    <div
      className={styles.rowGridContainer}
      style={{ gridTemplateColumns: `repeat(${React.Children.count(children)}, 1fr)` }}
    >
      {children}
    </div>
  );
}

TableHeader.propTypes = {
  children: PropTypes.node,
};

export default TableHeader;
