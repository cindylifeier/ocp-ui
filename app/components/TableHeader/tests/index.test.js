import React from 'react';
import { shallow } from 'enzyme';

import TableHeader from '../index';
import TableHeaderColumn from '../../TableHeaderColumn';

describe('<TableHeader />', () => {
  it('should match snapshot', () => {
    // Act
    const renderedComponent = shallow(
      (<TableHeader>
        <TableHeaderColumn>Col</TableHeaderColumn>
      </TableHeader>),
    );

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
});
