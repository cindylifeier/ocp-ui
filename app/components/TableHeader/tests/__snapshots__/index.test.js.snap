// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHeader /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableHeader>
    <TableHeaderColumn>
      Col
    </TableHeaderColumn>
  </TableHeader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <TableHeaderColumn>
        Col
      </TableHeaderColumn>,
      "className": undefined,
      "style": Object {
        "gridTemplateColumns": "repeat(1, 1fr)",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Col",
      },
      "ref": null,
      "rendered": "Col",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TableHeaderColumn>
          Col
        </TableHeaderColumn>,
        "className": undefined,
        "style": Object {
          "gridTemplateColumns": "repeat(1, 1fr)",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Col",
        },
        "ref": null,
        "rendered": "Col",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
